src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
+type Shape = 'Triangle' | 'Circle' | 'Rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const conditions = [
+      this.a <= 0,
+      this.b <= 0,
+      this.c <= 0,
+      this.a + this.b <= this.c,
+      this.b + this.c <= this.a,
+    ];
 
-export class Circle {
+    if (conditions.some((condition) => (condition))) {
```

Comment:
```suggestion
    if (conditions.some(Boolean)) {
```

Reply:
fixed, I didn`t use this sintaxis early, now I know, thanks